components/match2/wikis/teamfortress/match_group_input_custom.lua

```diff
@@ -0,0 +1,389 @@
+	-- Calculate and set winner, resulttype, placements and walkover (if applicable for the outcome)
+	elseif Logic.readBool(data.finished) then
+		if MatchGroupInput.isDraw(indexedScores) then
+			data.winner = 0
+			data.resulttype = 'draw'
+			indexedScores = MatchGroupInput.setPlacement(indexedScores, data.winner, 'draw')
+		elseif MatchGroupInput.hasSpecialStatus(indexedScores) then
+			data.winner = MatchGroupInput.getDefaultWinner(indexedScores)
+			data.resulttype = 'default'
+			if MatchGroupInput.hasForfeit(indexedScores) then
+				data.walkover = 'ff'
+			elseif MatchGroupInput.hasDisqualified(indexedScores) then
+				data.walkover = 'dq'
+			elseif MatchGroupInput.hasDefaultWinLoss(indexedScores) then
+				data.walkover = 'l'
+			end
+			indexedScores = MatchGroupInput.setPlacement(indexedScores, data.winner, 'default')
+		else
+			local winner
+			indexedScores, winner = MatchGroupInput.setPlacement(indexedScores, data.winner, nil, data.finished)
+			data.winner = data.winner or winner
+		end
+	end
+
+	--set it as finished if we have a winner
+	if Logic.isNotEmpty(data.winner) then
+		data.finished = true
+	end
+
+	return data, indexedScores
+end
+
+---@param opponents table[]
+---@param winner integer?
+---@param specialType string?
+---@param finished boolean|string?
+---@return table[]
+---@return integer?
+function MatchGroupInput.setPlacement(opponents, winner, specialType, finished)
+	if specialType == 'draw' then
+		for key, _ in pairs(opponents) do
+			opponents[key].placement = 1
+		end
+	elseif specialType == 'default' then
+		for key, _ in pairs(opponents) do
+			if key == winner then
+				opponents[key].placement = 1
+			else
+				opponents[key].placement = 2
+			end
+		end
+	else
+		local temporaryScore
+		local temporaryPlace = -99
+		local counter = 0
+		for scoreIndex, opp in Table.iter.spairs(opponents, MatchGroupInput.placementSortFunction) do
+			local score = tonumber(opp.score) or ''
+			counter = counter + 1
+			if counter == 1 and Logic.isEmpty(winner) then
+				if finished then
+					winner = scoreIndex
+				end
+			end
+			if temporaryScore == score then
+				opponents[scoreIndex].placement = tonumber(opponents[scoreIndex].placement) or temporaryPlace
+			else
+				opponents[scoreIndex].placement = tonumber(opponents[scoreIndex].placement) or counter
+				temporaryPlace = counter
+				temporaryScore = score
+			end
+		end
+	end
+
+	return opponents, winner
+end
+
+---@param tbl table[]
+---@param key1 integer
+---@param key2 integer
+---@return boolean
+function CustomMatchGroupInput.placementSortFunction(tbl, key1, key2)
+	local value1 = tonumber(tbl[key1].score) or -99
+	local value2 = tonumber(tbl[key2].score) or -99
+	return value1 > value2
+end
+
+--
+-- match related functions
+--
+
+---@param match table
+---@return table
+function matchFunctions.getBestOf(match)
+	match.bestof = Logic.emptyOr(match.bestof, Variables.varDefault('bestof', DEFAULT_BESTOF))
+	Variables.varDefine('bestof', match.bestof)
```

Comment:
This is not used on example page? Will it used?

Reply:
Currently the copypaste generator doesnt use `|bestof=` so I can adjust this